test/beforeHeadersSend.test.js

```diff
@@ -29,172 +32,241 @@ let setMock;
+            });
+        });
+
+        describe("redemption attempted", () => {
+            beforeEach(() => {
+                resetVars();
+                resetSpendVars();
+            });
+            test(`redemption header added (SEND_H2C_PARAMS = false), config.id = ${config.id}`, () => {
+                setSpendFlag(url.host, true);
+                setSpentUrl(url.href, false);
+                setReedemMethod(config.id);
+                workflow.__set__("SEND_H2C_PARAMS", false);
+                let redeemHdrs = beforeSendHeaders(details, url);
+                let reqHeaders = redeemHdrs.requestHeaders;
+                expect(getSpendFlag(url.host)).toBeFalsy();
+                expect(getSpendId([details.requestId])).toBeTruthy();
+                expect(getSpentUrl([url.href])).toBeTruthy();
+                switch (config.id) {
+                    case 1:
+                        expect(getSpentTab([details.tabId]) == url.href).toBeTruthy();
+                        break
+                    case 2:
+                        expect(getSpentTab([details.tabId]) == url.href).toBeFalsy();
+                        break
+                    default:
+                        throw Error(`Unhandled config.id value => ${config.id}`)
+                }
+                expect(reqHeaders).toBeTruthy();
+                let headerName = workflow.__get__("HEADER_NAME");
+                expect(reqHeaders[0].name === headerName).toBeTruthy();
+                expect(reqHeaders[0].value === b64EncodedTokenNoH2CParams).toBeTruthy();
+            });
+
+            test(`redemption header added (SEND_H2C_PARAMS = true), config.id = ${config.id}`, () => {
+                setSpendFlag(url.host, true);
+                setSpentUrl(url.href, false);
+                setReedemMethod(config.id);
+                let redeemHdrs = beforeSendHeaders(details, url);
+                let reqHeaders = redeemHdrs.requestHeaders;
+                expect(getSpendFlag(url.host)).toBeFalsy();
+                expect(getSpendId([details.requestId])).toBeTruthy();
+                expect(getSpentUrl([url.href])).toBeTruthy();
+                switch (config.id) {
+                    case 1:
+                        expect(getSpentTab([details.tabId]) == url.href).toBeTruthy();
+                        break
+                    case 2:
+                        expect(getSpentTab([details.tabId]) == url.href).toBeFalsy();
+                        break
+                    default:
+                        throw Error(`Unhandled config.id value => ${config.id}`)
+                }
+                expect(reqHeaders).toBeTruthy();
+                let headerName = workflow.__get__("HEADER_NAME");
+                expect(reqHeaders[0].name === headerName).toBeTruthy();
+                expect(reqHeaders[0].value === b64EncodedToken).toBeTruthy();
+            });
+        });
+    })
+
+function setReedemMethod(config_id) {
```

Comment:
As above, (possibly an artifact of old code) this could probably be handled using `workflow.__get__("REDEEM_METHOD")` because `setConfig` runs before each test.

Reply:
In this setup it does not help, hoping to address that with a general refactor of tests.